### ISSUE TITLE
We need additional structures

### ISSUE THREAD

**Issue by leorockway** [oilboi/Crafter#94]
We now have dungeons, but even though they have spawners and loot they are kind of boring. Villages, mine shafts, sunken ships or other places to explore would make the game much better.

**Comment by Verticen** [comment]
I imagine that these features will probably be relatively far into the future of development, though I would like to see chest generate in dungeons to make them worthwhile. Exploration may seem kinda dull right now just because there is only one 'biome' right now. OG beta minecraft didn't have many structures - had a bit of a theme of loneliness, so oil may choose to focus on other gameplay mechanics first. 

MineClone 2 does have abandoned mines generation I think, so maybe that code could be bolted on.

**Comment by leorockway** [comment]
@Verticen dungeons do generate chests and spawners now.

**Comment by tacotexmex** [comment]
Agreed. In contrast to how MCL2 structures are generated and spawned I’d like for Crafter to be able to use player-created prefabs for structures to a larger degree.

In underground tunnels and dungeons for example, prefab slices of rooms and tunnels can be made collaboratively in-world on the dev server creative realm. These prefabs then get exported and possibly optimized for inclusion in the game.

Besides being more fun and community-driven it will enable for massive variation in the game, when more people can work on it.
